nasa-pictures-stealer/src/main/java/nasapicturesstealer/application/service/NasaService.java

```diff
@@ -0,0 +1,72 @@
+package nasapicturesstealer.application.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import nasapicturesstealer.domain.entity.NasaCamera;
+import nasapicturesstealer.domain.entity.NasaFields;
+import nasapicturesstealer.domain.entity.NasaPicture;
+import nasapicturesstealer.infrastructure.repository.NasaPicturesRepository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class NasaService {
+
+  @Value("${nasa.url}")
+  private String nasaUrl;
+  
+  @Value("${nasa.key}")
+  private String nasaKey;
+  
+  private final NasaPicturesRepository nasaPictureRepository;
+  
+  private final RestTemplate restTemplate;
```

Comment:
It will be good to think about creating NasaClient and moving all the logic into it.